### DISCUSSION TITLE
Question: Setting FMU inputs in fmi2EvaluateME()

### DISCUSSION THREAD

**Discussion by hejersbo** [ThummeTo/FMIFlux.jl#90] [Q&A]
Hi,

This is not a bug report. Merely a question (and perhaps clarification of a my misunderstanding :-))

I would like to specify the FMU inputs when calling `fmiEvaluteME()` and I can see that the `fmiEvaluate()` has arguments `setValueReferences` and `setValues`. But is it possible to set an input function (like for `fmiSimulateME`) that is evaluated with the input time of the ODE solver. So e.g. one could have something like `sin(t)` being the input function for the FMU.

**Comment by hejersbo** [comment]
The reason for this is that I try to make a simple example - more or less replicating the simple_hybrid_ME example of the repo but with a different ME FMU. But I setup the neural FMU just like the example (using `fmiEvaluateME()`).

```
# NeuralFMU setup
numStates = fmiGetNumberOfStates(referenceFMU)

net = Chain(inputs -> fmiEvaluateME(referenceFMU, inputs),
            Dense(numStates, 16, tanh),
            Dense(16, 16, tanh),
            Dense(16, numStates))

neuralFMU = ME_NeuralFMU(referenceFMU, net, (tStart, tStop), Tsit5(); saveat=tSave);
```

**Comment by ThummeTo** [comment]
Sorry for the late reply, somehow I skipped that notification...

Yes, you can totally do that, see this test here:
https://github.com/ThummeTo/FMIFlux.jl/blob/568e585d607abe25e51b20a5c7fadb7be38807fa/test/hybrid_ME.jl#LL134C6-L134C6

Basically your chain would look like this:
```
function myInputFunction(t)
   return [sin(t)]
end

net = Chain(x-> referenceFMU(;x=x, u_refs=setVRs, u=myInputFunction(  getCurrentComponent(referenceFMU).t  )),
            Dense(numStates, 16, tanh),
            Dense(16, 16, tanh),
            Dense(16, numStates))
```

`getCurrentComponent(referenceFMU).t` equals the time instance of the ODE solver.
